forjadev/gun-organization
# [GUN-16] setup swagger

# [Feature] Swagger Documentation for API Routes

This pull request introduces the integration of Swagger for API documentation. The objective of this PR is to improve the developer experience and ease of use for our APIs.

Swagger is an open-source tool that helps developers design, build, document, and consume RESTful web services. By integrating Swagger, we can now have automatic, up-to-date documentation for all of our APIs.

In this PR, the following changes have been made:

1. **Swagger Integration**: The Swagger UI has been added to our project. It can be accessed at the `/swagger/index.html` endpoint when the server is running. This provides a user-friendly interface to explore and test our API endpoints.

2. **Documented the "Ping" Route**: The `GET /ping` route has been fully documented in Swagger. This includes a detailed description of what the route does, the request it expects, and the response it returns. This serves as an example of how to document the remaining routes in our API.

## Testing:

To test this feature, follow these steps:

1. Pull the changes from this branch into your local environment.
2. Start the server.
3. Navigate to `http://localhost:<PORT>/swagger/index.html` in your web browser.
4. You should see the Swagger UI with our API's documentation, including the `GET /ping` route.

## Screenshots:
![image](https://github.com/forjadev/gun-organization/assets/29252011/8708a4b7-40bc-421a-bcda-95d9f80a1f52)

Please review this PR and let me know if there are any questions or concerns.

---
This PR is related to issue #16
